src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -22,16 +25,14 @@
     //Groups of 2: (option name) (option value)
     private static final Pattern ARGS_PATTERN = Pattern.compile("-(\\w+)\\s([^-]+)");
     private final Hashtable<String, String> parameters = new Hashtable<>();
-    private ArrayList<String> params = new ArrayList<>();
-
     private boolean exit = false;
 
 
-    public String parser(String userInput, ProjectManager projectListManager) {
+    public Command parser(String userInput, ProjectManager projectListManager) {
         if (userInput.equals(BYE)) {
             System.out.println(BYE);
             exit = true;
-            return null;
+            System.exit(0);
```

Comment:
Can you remove this line? This is because _line 34_ has set `exit = true`. In `Duke.java` _line 63_, I have already added the check to check if the `exit = true`. Also, the `Duke.java` has to call `sm.save()` before it exits.

Reply:
Ok that makes sense